apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs

```diff
@@ -74,13 +83,11 @@ public async Task Invoke(HttpContext httpContext, TraceHeaderContext traceHeader
             }
             else
             {
-                if (traceHeaderContext.TraceId != null)
+                if (traceContext.TraceId != null)
                 {
-                    // Set the trace updated trace header on the response.
-                    var updatedHeaderContext = TraceHeaderContext.Create(
-                        tracer.GetCurrentTraceId(), tracer.GetCurrentSpanId() ?? 0, true);
-                    httpContext.Response.Headers.Add(
-                        TraceHeaderContext.TraceHeader, updatedHeaderContext.ToString());
+                    // Set the current trace context on the response.
+                    var currentTraceContext = ContextTracerManager.GetCurrentTraceContext();
+                    traceContextPropagator(httpContext.Response, currentTraceContext);
```

Comment:
Using `traceContextPropagator.Invoke` might be clearer than just `traceContextPropagator(...)`. Not sure.